File: src/AutoRest.SdkExplorer/Model/Schema/SchemaMethod.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;
using System.Linq;

namespace AutoRest.SdkExplorer.Model.Schema
{
    public class SchemaMethod
    {
        public string Name { get; set; } = string.Empty;

        public List<SchemaMethodParameter> MethodParameters { get; set; } = new List<SchemaMethodParameter>();

        public SchemaMethod()
        {

        }

        public bool HasParameter(string parameterSerializerName)
        {
            return this.MethodParameters.Any(mp => mp.RelatedPropertySerializerPath == parameterSerializerName);
        }
    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Schema/SchemaMethod.ts
import { SchemaMethodParameter } from "./SchemaMethodParameter";

export class SchemaMethod implements AutoRest.SdkExplorer.Interface.SchemaMethod {
    name: string;
    methodParameters: SchemaMethodParameter[];

    constructor(data: AutoRest.SdkExplorer.Interface.SchemaMethod) {
        this.name = data.name!;
        this.methodParameters = data.methodParameters?.map(d => new SchemaMethodParameter(d)) ?? [];
    }
}
